src/App/Kernel.php

```diff
@@ -138,14 +142,21 @@ public function dependencies(Application $app)
                     // Configure discriminators and subscribers here.
                     $dispatcher->addSubscriber(new RecommendationResultDiscriminator());
                 })
-                ->setCacheDir(self::ROOT.'/var/cache')
+                ->setCacheDir(self::CACHE_DIR)
                 ->build();
         };
         $app['serializer.context'] = function () {
             return SerializationContext::create();
         };
         // Puli.
-        $app['puli.factory'] = function () {
+        $app['puli.factory'] = function (Application $app) {
+            if ($app['debug'] && defined('PULI_FACTORY_CLASS') === false) {
+                throw new LogicException('
+                    Puli cannot be found in your composer auto-loader, please downgrade composer to 
+                    composer 1.0.0 beta 2 (`composer self-update 1.0.0-beta2`) and your puli to 
```

Comment:
we use this Composer, and it usually works: https://github.com/elifesciences/builder-base-formula/blob/master/elife/puli.sls#L5 (better than a beta)